### DISCUSSION TITLE
Feature Request: Full screening a window should mean that the other windows shouldnt be visible at all

### DISCUSSION THREAD

**Discussion by skela** [qtile/qtile#3610] [Ideas]
By fullscreen i mean the `qt.current_window.toggle_fullscreen()` function.

I have semi-transparency in my terminals so when i have say 3 or 4 terminals (or other programs) visible and i want to focus on 1 particular terminal in fullscreen, its a bit distracting seeing the other windows below my fullscreen window.

Maybe its already possible and i dont know how to do it? But in any case I would argue that this would be a really cool feature to make qtile even better than it is today.
![2022-06-04_21-44](https://user-images.githubusercontent.com/58268/172023630-3be3b717-3d0b-41fb-8b54-0055492234cf.png)
![2022-06-04_21-54](https://user-images.githubusercontent.com/58268/172023632-99ac6a1e-c25b-4cb1-8243-e3df480e90ab.png)

**Comment by m-col** [comment]
> Maybe its already possible and i dont know how to do it? 

You can't, per se. But what you *can* do is, intead of fullscreening, use the `Max` layout and hide the bar. I think that might get close to what you want? You can pass multiple commands to `Key` so you can do something like this:
```py
Key([mod], "f", lazy.next_layout(), lazy.hide_show_bar("top"))
```
The `Max` layout hides all of the other windows for you. If you use one other layout, this command will toggle betweent the two states for you. If you already use multiple layouts then you might need to use a wrapper function to jump between layouts, toggling the bar.

**Reply by skela** [reply]
ooh, thats an excellent idea.
is there really no way to jump to a specific layout?

**Reply by skela** [reply]
I do the shortcut triggering a function, have tried to do tricks with hiding the bar before so thats cool, but i think for your idea to work, it really would require a way to change to a specific layout (max), then would need to maybe remember the layout it was in prior to changing so i could go back to the layout before i went fullscreen.

![2022-06-04_22-27](https://user-images.githubusercontent.com/58268/172024605-3f14deba-00fd-43f8-8acf-0893393682e8.png)

**Reply by m-col** [reply]
`Group` has a `layout` property you can use:
```py
qt.current_group.layout = "max"
```
One way you could store the state for later restoration might be a dictionary, so you can have an item per group:
```py
real_layout = {}

def toggle_fullscreen(qtile):
    group = qtile.current_window.group
    if group in real_layout:
        group.layout = real_layout.pop(group)
    else:
        real_layout[group] - group.layout.name
        group.layout = "max"
```
(untested!)

**Reply by skela** [reply]
@m-col thx mate, thats really amazing. Worked with a bit of tweaks, I think actually this is a really awesome solution, because it means that out of the box I can use my other shortcuts to quickly switch between the active full screen window (because its using the max layout when in full screen).
![2022-06-04_22-57](https://user-images.githubusercontent.com/58268/172025271-954b44ce-97ca-4d76-b43a-569f164d4313.png)

**Reply by m-col** [reply]
Nice, glad you got it working :)

**Reply by skela** [reply]
tweaked it a bit more - this works really well.

![2022-06-04_23-00](https://user-images.githubusercontent.com/58268/172025349-1055e94c-e411-41a6-9f9b-f23d3a0e9e91.png)
